### ISSUE TITLE
Possible bug when upgrading to 0.7.0 and openid_connect to 2.2.0

### ISSUE THREAD

**Issue by raivil** [omniauth/omniauth_openid_connect#156]
The method `fetch_key` is trying to JSON parse a Hash, causing the error `unexpected character '=' [0x3d] (after keys) at line 1, column 8 [sparse.c:776] (JSON::ParserError)`

```
 def fetch_key
    @fetch_key ||= parse_jwk_key(::OpenIDConnect.http_client.get(client_options.jwks_uri).body)
  end
```
This might be because `OpenIDConnect.http_client` is now using Faraday from upgrading openid_connect to 2.2.0 and automatically converting json to hash (https://github.com/nov/openid_connect/blob/master/lib/openid_connect.rb#L72) and this recent change https://github.com/omniauth/omniauth_openid_connect/pull/153/files#diff-ee829bbbbd9d35ee93126d96347fffbfa8756868001c1d4ad6aff9c4650e8151R231

Manual reproduction of the error:
```
 x = ::OpenIDConnect.http_client.get("http://auth.lvh.me:3000/oauth/discovery/keys").body
=> 
{"keys"=>                 
...                       
irb(main):008:0> x.class
=> Hash
irb(main):009:0> JSON.parse(x)
(irb):9:in `parse': unexpected character '=' [0x3d] (after keys) at line 1, column 8 [sparse.c:776] (JSON::ParserError)
```

```
irb(main):011:0> ::OpenIDConnect.http_client
=> 
#<Faraday::Connection:0x0000000112a55510
 @builder=#<Faraday::RackBuilder:0x0000000112a55150 @adapter=Faraday::Adapter::NetHttp, @handlers=[Faraday::Request::UrlEncoded, Faraday::Request::Json, Faraday::Response::Json]>,
 @default_parallel_manager=nil,
 @headers={"User-Agent"=>"OpenIDConnect (2.2.0)"},
 @manual_proxy=false,
 @options=#<struct Faraday::RequestOptions params_encoder=nil, proxy=nil, bind=nil, timeout=nil, open_timeout=nil, read_timeout=nil, write_timeout=nil, boundary=nil, oauth=nil, context=nil, on_data=nil>,
 @parallel_manager=nil,
 @params={},
 @proxy=nil,
 @ssl=
  #<struct Faraday::SSLOptions
   verify=false,
   verify_hostname=nil,
   ca_file=nil,
   ca_path=nil,
   verify_mode=nil,
   cert_store=nil,
   client_cert=nil,
   client_key=nil,
   certificate=nil,
   private_key=nil,
   verify_depth=nil,
   version=nil,
   min_version=nil,
   max_version=nil>,
 @url_prefix=#<URI::HTTP http:/>>
```

**Comment by stanhu** [comment]
Thanks, https://github.com/omniauth/omniauth_openid_connect/pull/157 should fix this.

**Comment by stanhu** [comment]
Released v0.7.1 to address this.
